components/net/image_cache.rs

```diff
@@ -405,14 +401,39 @@ impl ImageCacheImpl {
     }
 }
 
+struct ImageDecoderRunnable {
```

Comment:
I don't see a reason for this separate structure rather than having the code inline in the caller of `run`.